src/shrub/v3/shrub_service.py

```diff
@@ -14,8 +117,11 @@ def generate_yaml(shrub_config: BaseModel) -> str:
         :param shrub_config: Shrub configuration to generate.
         :return: YAML version of given shrub configuration.
         """
-        return yaml.safe_dump(
-            shrub_config.dict(exclude_none=True, exclude_unset=True, by_alias=True)
+        return yaml.dump(
```

Comment:
Is this a breaking change (`safe_dump` doesn't support custom objects IIRC)?

Reply:
The new class subclasses `yaml.SafeDumper`, so it should behave the same.  I'll add a test to confirm.

Reply:
Thanks for confirm this, another question regarding backwards compatibility, would this dump the same yaml as before? I'm asking because if it's not fully backwards compatible, we should bump the version to `4.0.0`, otherwise `3.2.0` since this is a new feature.

Reply:
As I understand it, the yaml content will be the same, we are purely changing the formatting to make it more human-friendly and align with how most hand-written configs are formatted.  I'd prefer 3.2.0.